### PR TITLE
fixing docs

### LINKED ISSUE #378: docs failing
Our documentation build started failing about two weeks ago on readthedocs.io. Error message:
```
Searching for python-igraph>=0.7.1
Reading https://pypi.org/simple/python-igraph/
Downloading https://files.pythonhosted.org/packages/0f/a0/4e7134f803737aa6eebb4e5250565ace0e2599659e22be7f7eba520ff017/python-igraph-0.7.1.post6.tar.gz#sha256=a5ea42790a077eadb3d1e8f82edc47516fea4a2b26bbf52bfc148370fe0fe676
Best match: python-igraph 0.7.1.post6
Processing python-igraph-0.7.1.post6.tar.gz
Writing /tmp/easy_install-x9nn5yfc/python-igraph-0.7.1.post6/setup.cfg
Running python-igraph-0.7.1.post6/setup.py -q bdist_egg --dist-dir /tmp/easy_install-x9nn5yfc/python-igraph-0.7.1.post6/egg-dist-tmp-iv798fpq
Cannot find the C core of igraph on this system using pkg-config.
We will now try to download and compile the C core from scratch.
Version number of the C core: 0.7.1.post6
We will also try: 0.7.1

error: Setup script exited with error: SandboxViolation: mkdir('/home/docs/checkouts/readthedocs.org/user_builds/scvi/checkouts/master/tmp', 511) {}

The package setup script has attempted to modify files on your system
that are not within the EasyInstall build area, and has been aborted.

This package cannot be safely installed by EasyInstall, and may not
support alternate installation locations even if you run its setup
script by hand.  Please inform the package's author and the EasyInstall
maintainers to find out if a fix or workaround is available.
```

### PR DESCRIPTION
**Pull request by romain-lopez**
changed setup.py and added extras

fixes #378 and #336

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -27,16 +27,14 @@
...
-    "python-igraph>=0.7.1",
     "hyperopt>=0.1.2",
 ]
 
 setup_requirements = ['pytest-runner', ]
 
 test_requirements = ['pytest', ]
 
-extras_requirements = {'test': ['scanpy', 'louvain', ]}
+extras_requirements = {'test': ['scanpy', 'louvain', "leidenalg>=0.7.0", "python-igraph>=0.7.1", ]}
```
Comment (jeff-regier) [inline setup.py]: Do we even need to list them as extra requirements now that you've removed `utils.py`?

**Comment by jeff-regier** [comment]
How does this address #336 ?

File: setup.py (hunk 1 above)
Comment (romain-lopez) [reply setup.py]: I think we need them to make work scanpy but not sure exactly

**Comment by romain-lopez** [comment]
This answer to the last problem in #336, not the whole issue which was already fixed

File: setup.py (hunk 1 above)
Comment (jeff-regier) [reply setup.py]: scanpy really should include it's own dependencies, but whatever, let's close this. Thanks for making the changes.

**Pull request merged**